src/model/AirLoopHVACUnitaryHeatPumpAirToAir.cpp

```diff
@@ -388,28 +392,41 @@ namespace model {
           // ! hvacComponent.optionalCast<FanSystemModel>()
       ) {
         return false;
-      };
+      }
 
       return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::SupplyAirFanName, hvacComponent.handle());
-      ;
     }
 
     bool AirLoopHVACUnitaryHeatPumpAirToAir_Impl::setHeatingCoil(HVACComponent& hvacComponent) {
-      if (!hvacComponent.optionalCast<CoilHeatingDXSingleSpeed>()) {
-        return false;
-      };
+      bool isTypeOK = false;
 
-      return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::HeatingCoilName, hvacComponent.handle());
-      ;
+      if (hvacComponent.optionalCast<CoilHeatingDXSingleSpeed>()) {
+        isTypeOK = true;
+      } else if (hvacComponent.optionalCast<CoilHeatingDXVariableSpeed>()) {
+        isTypeOK = true;
+      }
```

Comment:
This should be unnecessary, the IDD ref will enforce that it's the correct type. I suggest removing the check, less harcoding in code means more flexibility if a type is supported later (which probably won't happen, given that this object should be superseded by AirLoopHVACUnitarySystem I think)

Reply:
Edit: actually, this is correct. I didn't realize initially that the `OS:Coil:Heating:DX:VariableRefrigerantFlow` is also a `HeatingCoilsDXSingleSpeed`, but the I/O ref is clear https://bigladdersoftware.com/epx/docs/9-4/input-output-reference/group-unitary-equipment.html#airloophvacunitaryheatpumpairtoair

> Heating coil type must be either Coil:Heating:DX:SingleSpeed or Coil:Heating:DX:VariableSpeed.